### ISSUE TITLE
Projects are not summarized/built/indexed unless you edit the "top-file"

### ISSUE THREAD

**Issue by jurgenvinju** [usethesource/rascal-language-servers#243]
**Describe the bug**

If I just want to browse a Rascal project with jump-to-def as the main link provider, this does not work unless
I edit the topmost module in a project to trigger a full build.

The reason is that there is no trigger for the type-checker to run other than an edit action. I find this contra-intuitive myself and I also receive bug reports from people claiming that the IDE is broken. Which it isn't really.

**Comment by DavyLandman** [comment]
I've also observed it also hurts when there are multiple projects in play. So A imports B, but B is in a different project with an empy target folder.

**Comment by jurgenvinju** [comment]
Yes. We do not trigger the type-checker across project reverse-dependencies (like Eclipse does automagically), and so you have to trigger it manually for every project you depend on.

**Comment by DavyLandman** [comment]
This has been solved, we now automatically find the dependency graph and if it's in the same project, rascal-typechecker will figure it out. if it's across projects, we schedule the other projects in the right order.
